apps/researcher/src/components/form/edtf/converter.ts

```diff
@@ -0,0 +1,36 @@
+export function toDateString(yyyy = '', mm = '', dd = ''): string {
+  if (yyyy === '' && mm === '' && dd === '') {
+    return '';
+  }
+
+  const yyyyPadded =
+    yyyy.charAt(0) === '-'
+      ? `-${yyyy.substring(1).padStart(4, '0')}`
+      : yyyy.padStart(4, '0');
+  const mmPadded = mm.padStart(2, '0');
+  const ddPadded = dd.padStart(2, '0');
+
+  if (ddPadded === '00' && mmPadded === '00') return yyyyPadded;
+  if (ddPadded === '00') return `${yyyyPadded}-${mmPadded}`;
+  if (mmPadded === '00') return yyyyPadded;
+  return `${yyyyPadded}-${mmPadded}-${ddPadded}`;
+}
+
+export function fromDateString(dateString: string): {
+  yyyy: string;
+  mm: string;
+  dd: string;
+} {
+  const parts = dateString.split(/(?<=^-?\d{1,4}|\d{2})-/);
```

Comment:
In the unit tests these `parts` seem to be called `components`. Use one term, for consistency?

Reply:
Good point, I will go for `parts`